jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -329,43 +321,43 @@ public Runnable onIdleTimeout(TimeoutException t)
             if (LOG.isDebugEnabled())
                 LOG.debug("onIdleTimeout {}", this, t);
 
-            // if not already a failure,
-            if (_failure == null)
+            Runnable invokeOnContentAvailable = null;
+            if (_readFailure == null)
             {
-                // if we are currently demanding, take the onContentAvailable runnable to invoke below.
-                Runnable invokeOnContentAvailable = _onContentAvailable;
+                // If there is demand, take the onContentAvailable runnable to invoke below.
+                invokeOnContentAvailable = _onContentAvailable;
                 _onContentAvailable = null;
 
-                // If demand was in process, then arrange for the next read to return the idle timeout, if no other error
+                // If there was demand, then arrange for the next read to return a transient chunk failure.
                 if (invokeOnContentAvailable != null)
-                    _failure = Content.Chunk.from(t, false);
+                    _readFailure = Content.Chunk.from(t, false);
+            }
 
-                // If a write call is in progress, take the writeCallback to fail below
-                Runnable invokeWriteFailure = _response.lockedFailWrite(t);
+            // If a write call is pending, take the writeCallback to fail below.
+            Runnable invokeWriteFailure = _response.lockedFailWrite(t, false);
```

Comment:
the `lockFailWrite` method does not handle the case that there is already a write failure.